backtracking/all_subsequences.py

```diff
@@ -1,35 +1,53 @@
 """
-In this problem, we want to determine all possible subsequences
-of the given sequence. We use backtracking to solve this problem.
+In this problem, we find all subsequences of a given list using backtracking.
+Time Complexity: O(2^n), where n is the list's length.
 
-Time complexity: O(2^n),
-where n denotes the length of the given sequence.
+Functions:
+- generate_all_subsequences: Initiates backtracking and prints subsequences.
+- create_state_space_tree: Recursively builds a state-space tree for subsequences.
 """
-from __future__ import annotations
 
+from __future__ import annotations
 from typing import Any
 
 
```

Comment:
check the `Check with  `pre-commit run --all-files --show-diff-on-failure` to know more about your pre commits !